react/src/components/atoms/buttons/ButtonWithIcon/index.js

```diff
@@ -43,12 +43,21 @@ class ButtonWithIcon extends React.Component {
       onClick: this.handleClick,
       tabIndex: 0
     };
-    return(
-      <button {...buttonProps} aria-label={this.props.ariaLabel} ref={this.setButtonRef}>
-        <span>{this.props.text}</span>
-        {this.props.icon}
-      </button>
-    );
+    if (this.props.ariaLabel !== '') {
```

Comment:
This could be simplify if you pull the ariaLabel out and assign it to buttonProps,
e.g. this.props.ariaLabel && buttonProps['aria-label'] = this.props.ariaLabel
then you don't have do it conditional rendering below

Reply:
Also please add arialLabel in story knob